xcdat/spatial.py

```diff
@@ -738,7 +731,17 @@ def _averager(self, data_var: xr.DataArray, axis: List[SpatialAxis]):
         """
         weights = self._weights.fillna(0)
 
+        # The axis's dimension name is extracted directly from its related
+        # coordinate variable. This is is done in case the `axis` or
+        # `standard_name` attr is not set on the coord var for `cf_xarray` to
+        # interpret. For example, if `axis=["Y"]`, the dim name "lat" is
+        # extracted from `self._Y` (a DataArray coord var).
+        dim_names = []
+        for key in axis:
+            dim_name = getattr(self, f"_{key}").name
+            dim_names.append(dim_name)
+
         with xr.set_options(keep_attrs=True):
-            weighted_mean = data_var.cf.weighted(weights).mean(axis)
+            weighted_mean = data_var.cf.weighted(weights).mean(dim=dim_names)
```

Comment:
`cf_xarray` was attempting to interpret the dimension names using the `"axis"` attr on the coordinate variables. However, this breaks if the `"axis"` attr is not set.

Instead, we get the dimension names directly from the coordinate variable.